xarray/core/common.py

```diff
@@ -904,6 +909,9 @@ def coarsen(
         core.rolling.DataArrayCoarsen
         core.rolling.DatasetCoarsen
         """
+        if keep_attrs is None:
+            keep_attrs = _get_keep_attrs(default=False)
```

Comment:
Do we then need to pass this into `_coarsen_cls` and handle it there?

Reply:
I will try

Reply:
Cheers! Let us know if you need any guidance!